pkg/ccl/utilccl/license_check.go

```diff
@@ -35,8 +36,36 @@ var enterpriseLicense = settings.RegisterStringSetting(
 	"",
 	settings.WithValidateString(
 		func(sv *settings.Values, s string) error {
-			_, err := decode(s)
-			return err
+			l, err := decode(s)
+			if err != nil {
+				return err
+			}
+			if l == nil {
+				return nil
+			}
+
+			reportingSetting, ok, _ := settings.LookupForLocalAccess("diagnostics.reporting.enabled", true /* forSystemTenant */)
```

Comment:
nit: this looks like an non-standard way of fetching the setting. could we use `logcrash.DiagnosticsReportingEnabled.Get(sv)` , so that we avoid the unnecessary parsing/decoding steps?

Reply:
Ah I see, since this is a backport - should that be done into main and then cherry-picked to this PR?

Reply:
Going to reopen this in another PR